src/ghdcbot/engine/notifications.py

```
@@ -0,0 +1,315 @@
+) -> bool:
+    """Send notification via DM or channel. Returns True if sent."""
+    if not policy.allow_discord_mutations:
+        logger.debug("Skipping notification: Discord writes disabled (dry-run/observer)")
+        return False
+    
+    if channel_id:
+        send_msg = getattr(discord_writer, "send_message", None)
+        if callable(send_msg):
+            try:
+                send_msg(channel_id, message)
+                return True
```

Ruff (0.14.14)

[warning] Consider moving this statement to an `else` block

(TRY300)